### PR TITLE
feat: config driven trainers

### PR DESCRIPTION
**Pull request by nkoppisetty**
Explain what this PR does.
1. Updated main, preprocess and threshold model trainers to use config
2. Updated postprocess to use config, and generate max/weighted unified score

### CODE REVIEW

File: numaprom/config/default/numalogic.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 model:
   name: "SparseVanillaAE"
   conf:
-    seq_len: 12
+    seq_len: 10
```
Comment (ab93) [inline numaprom/config/default/numalogic.yaml:4]: Let's keep it 12?

File: numaprom/udsink/train_rollout.py [hunk 2]
```diff
@@ -58,14 +59,14 @@ def clean_data(uuid: str, df: pd.DataFrame, hash_col: str, limit=12) -> pd.DataF
     return df
 
 
-def _train_model(uuid, x, anomaly_model):
+def _train_model(uuid, x, model_info, trainer_cfg):
     _start_train = time.perf_counter()
 
-    win_size = anomaly_model.conf["seq_len"]
+    win_size = model_info.conf["seq_len"]
     dataset = StreamingDataset(x, win_size)
```
Comment (ab93) [inline numaprom/udsink/train_rollout.py:66]: Can you use ModelFactory here? https://github.com/numaproj/numalogic/blob/main/numalogic/config/factory.py#L69

File: numaprom/udsink/train_rollout.py [hunk 3]
```diff
@@ -76,16 +77,24 @@ def _train_model(uuid, x, anomaly_model):
     return train_reconerr.numpy(), model
 
 
-def _preprocess(x_raw):
+def _preprocess(x_raw, preproc_cfg: List[ModelInfo]):
     clf = StandardScaler()
```
Comment (ab93) [inline numaprom/udsink/train_rollout.py:81]: You can remove this

**Review by ab93 — COMMENTED** [review]
Left a few comments

File: numaprom/config/default/numalogic.yaml (hunk 1 above)
Comment (ab93) [reply numaprom/config/default/numalogic.yaml:4]: Any reason why we made it 10?

File: numaprom/udsink/train_rollout.py [hunk 4]
```diff
@@ -139,19 +145,24 @@ def train_rollout(datums: List[Datum]) -> Responses:
             responses.append(Response.as_success(_datum.id))
             continue
 
-        if len(train_df) < win_size:
+        if len(train_df) < model_cfg.conf["seq_len"]:
```
Comment (ab93) [inline numaprom/udsink/train_rollout.py:148]: I think we can replace this with checking against MIN_TRAIN_SIZE. And maybe replace this with the check we have in `clean_data()`. We can remove the check there. What do you think?

**Review by ab93 — COMMENTED** [review]
Added a comment

File: numaprom/udsink/train_rollout.py [hunk 5]
```diff
@@ -131,27 +129,32 @@ def train_rollout(datums: List[Datum]) -> Responses:
...
-        if len(train_df) < win_size:
-            _LOGGER.info(
-                "%s - Skipping training since traindata size: %s is less than winsize: %s",
+        if len(train_df) < MIN_TRAIN_SIZE:
+            _LOGGER.warning(
+                "%s - Skipping training, train data less than minimum required: %s, df shape: %s",
                 payload.uuid,
+                MIN_TRAIN_SIZE,
                 train_df.shape,
-                win_size,
```
Comment (ab93) [inline numaprom/udsink/train_rollout.py:147]: i think this is an extra param in the log.

Comment (nkoppisetty) [reply numaprom/udsink/train_rollout.py:147]: it matches, 3 params

**Review by ab93 — APPROVED** [review]

**Pull request merged**